R/edgeThresh.R

```diff
@@ -0,0 +1,461 @@
+#'
+#'        ( )     ( )                  ( )  ←  ( )
+#'         ↑       ↓             →
+#'        (A)  →  (B)                  (A)   ⇉  (B)
+#'
+#'     "Self-edges": e.g. (A ↔ B, C ↔ A) to (A ↔ A, B ↔ C)
+#'     Where in the case of:
+#'     'Unidirectional edges': Precedent of one edge is the antecedent of the
+#'     other edge to be swapped.
+#'     'Bidirectional edges': The edge pair to be swapped involves less than 4
+#'     distinct nodes.
+#'     "Disconnected components": Considering (A → B, C → D), since A and B are
+#'     in the same component, they should still be in the same component when
+#'     (A → D, C → B). This guarantees that the component remains connected.
+#'
+#'     Note that (A → B, A → D) to (A → D, A → B) is not considered a swap, as
+#'     an edge swap has to involve 4 distinct vertices. This will also avoid
+#'     multiple edges and self-edges, as they involve less than 4 distinct
+#'     vertices.
+#'
+#' @section Maintaining Connected Components:
+#'     Note: In order to minimize computation, and since the target outcome is
+#'     to measure order of the largest connected component, only the largest
+#'     connected component of the input is then used for the computation in case
+#'     the input has more than one component.
+#'
+#'     To ensure nodes maintain their components during permutation,
+#'     the size and distribution of the components of the graph is measured
+#'     initially, and and only the largest connected component is permutated.
+#'
+#'     Then the strongly connected components within the largest component are
+#'     found.
+#'     After each swap, the new component of the sending endpoint (A in A → B)
+#'     is the same as the component the receiving endpoint belongs to. Thus
+#'     the component maintenance is achieved in 2 steps (one query and one
+#'     comparison) for each receiving node, and thus 4 * Q * |E| steps are
+#'     required.
+#'
+#' @section Incrementing δ and Lmax:
+#'     After creating the permutated graph, the following procedure is run on
+#'     the graph for each Lmax starting from Lmin to the Lmax arguments.
+#'     Initially, δ values are chosen by finding min(|E|, 100) quantiles of
+#'     edge weight distribution, and used to delete edges from the permutation.
+#'     As soon as the largest component has size (ord) between Lmin and Lmax,
+#'     If the current permutation's ord values is less than Lmin (without
+#'     finding δ), the delta is skipped, and interpolated later on.
+#'
+#' @param EGG An EGG object - the input heat-equilibrated gene score graph.
+#' @param Lmax	Maximum order for the largest connected component in a random
+#' network. See details.
+#' @param Lmin Minimum order for the largest connected component. Defaults to 2.
+#' @param N	The number of permuted networks to produce for choosing a δ
+#' cutoff.
+#' @param Q	The number of edge-swapping permutations to produce one of the
```

Comment:
@RicardoRam This is what the design enforces, can I change what is specified on the task page?